add dark mode and interaction

tcase360 opened the pull request.

tcase360 committed:
add dark mode and interaction

AetherUnbound reviewed (approved):
Definitely wait for @sarayourfriend's review on this, but the preview looks amazing! Great job!

sarayourfriend commented on main.js:
```diff
@@ -0,0 +1,34 @@
+        const ariaPressed = document.body.classList.contains('light') ? false : true;
+        
+        themeToggleButton.setAttribute('aria-pressed', ariaPressed);
+        localStorage.setItem('theme', theme);
```
Nice idea storing it in local storage!

sarayourfriend commented on main.js:
```diff
@@ -0,0 +1,34 @@
+        if (e.key === " " || e.key === "Enter" || e.key === "Spacebar") { // "Spacebar" for IE11 support
+            // Prevent the default action to stop scrolling when space is pressed
+            e.preventDefault();
+            toggleButton(e.target);
```
Where does `toggleButton` come from?

tcase360 replied:
Whoops, was meant to be a function call to the toggle function for the correct keys. Fixed.

tcase360 committed:
fix theme toggle with variables and clean up code

tcase360 committed:
add accessibility support

sarayourfriend commented on _includes/style.scss:
```diff
@@ -1,19 +1,34 @@
 }
 
+a {
+    color: #ff0080;
```
```suggestion
    color: $secondary-color;
```

sarayourfriend commented on _includes/style.scss:
```diff
@@ -156,48 +176,27 @@ h3 {
+        }
+
+        .phone-icon {
+            background-image: url('icons/phone-light.svg');
```
Should these URLs be absolute, just in case?
```suggestion
            background-image: url('/icons/phone-light.svg');
```

tcase360 replied:
This is outdated now, not using urls

sarayourfriend commented on main.js:
```diff
@@ -0,0 +1,39 @@
+    const themeToggleButton = document.querySelector('.theme-toggle');
+    const darkSchemePreference = prefersDarkScheme ? 'dark' : 'light';
+    const currentTheme = localStorage.getItem('theme') ? localStorage.getItem('theme') : darkSchemePreference;
+    const rootElement = document.querySelector(':root');
```
It looks like `:root` resolves to the `html` element instead of the `body`. I wonder if we should just use `body` everywhere we're using root? The class doesn't seem to be applying correct onto the `html` element.

sarayourfriend replied:
Wait never mind I think it's still just the issue where it doesn't work if you start off from light.

tcase360 committed:
move phone icon from css background to svg and manipulate fill/color

sarayourfriend commented on _includes/resource.html:
```diff
@@ -5,7 +5,9 @@ <h4><a href="{{ resource.href }}">{{ resource.title }}</a></h4>
         {% if resource.phones.primary %}
-            <a href="tel:{{ resource.phones.primary.link }}"><span class="phone-icon" role="img" aria-label="Phone"></span></a>
+            <a href="tel:{{ resource.phones.primary.link }}"><span class="phone-icon" role="img" aria-label="Phone">
+                <svg xmlns="http://www.w3.org/2000/svg" class="phone-light" width="24" height="24" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-phone-call"><path d="M15.05 5A5 5 0 0 1 19 8.95M15.05 1A9 9 0 0 1 23 8.94m-1 7.98v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07 ...
```
To avoid repeating the same glyph all over the page I wonder if we could define it once and then use the [`use`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use) element to reference it? I believe it'll still be styleable.

I've never tried this before though, but it seems wrong to repeat the glyph over and over again on the page.

AetherUnbound replied:
Do you think we should just define it once as a hidden SVG and then just reference the definition on each resource? I'm not sure how we'd do it easily otherwise

sarayourfriend replied:
Yeah that'd be the idea I think.

sarayourfriend replied:
Although I wonder if we could just have `phone-light` and `phone-dark` and then use CSS selectors to toggle one or the other as `display: none` depending on the selected color scheme? Then we can render both as `img src="/whatever"` instead of inlining the SVG.

AetherUnbound replied:
Oh that makes sense! I can play around with that - do you mind if I do that in a follow up PR though? I'd love to get this merged & the repo transferred and I don't want it contingent on my ability to mess with the proper display modes :sweat_smile:

sarayourfriend replied:
Yup sounds good!

AetherUnbound committed:
Use correct equality check, pull matches attribute from mediaquery

AetherUnbound committed:
Reference secondary color appropriately

AetherUnbound committed:
Always apply light/dark conditional style

The current SCSS setup had the light/dark conditional styling *only* apply when the media query determined the browser prefers dark mode. Otherwise it was just ignored. Since we're already using that media query in main.js to determine the style, we don't need to have it here.

AetherUnbound committed:
Update gemfile.lock

sarayourfriend reviewed (approved):
LGTM!

AetherUnbound commented:
Got it!

![image](https://user-images.githubusercontent.com/10214785/131739882-7641825b-fbe8-485e-a8b6-8288467a0ff5.png)

AetherUnbound committed:
Make button shadow & border boxes respect theme
